pkg/kube/kroler_test.go

```diff
@@ -0,0 +1,67 @@
+package kube
+
+import (
+	"context"
+	"testing"
+
+	"github.com/ash2k/iam4kube"
+	"github.com/ash2k/iam4kube/pkg/util/logz"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/stretchr/testify/assert"
+	core_v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+func newKroler() Kroler {
+	store := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{podByIpIndex: podByIpIndexFunc})
+	store2 := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{})
+
+	kroler := Kroler{
+		logger:    logz.DevelopmentLogger(),
+		podIdx:    store,
+		svcAccIdx: store2,
+	}
+	return kroler
+}
+
+func TestRoleForIpNotFound(t *testing.T) {
+	t.Parallel()
+
+	kroler := newKroler()
+	_, err := kroler.RoleForIp(context.Background(), "127.0.0.1")
+	assert.Equal(t, err, ErrPodForIpNotFound)
+}
+
+func TestRoleForIp(t *testing.T) {
+	t.Parallel()
+
+	namespace := "Foo"
+	name := "Bar"
+	ipAddr := "127.0.0.1"
+
+	kroler := newKroler()
+
+	pod := core_v1.Pod{
+		Spec:   core_v1.PodSpec{ServiceAccountName: name},
+		Status: core_v1.PodStatus{PodIP: ipAddr},
+	}
+	pod.Namespace = namespace
+	kroler.podIdx.Add(&pod)
+
+	svcAccount := core_v1.ServiceAccount{}
+	svcAccount.Name = name
+	svcAccount.Namespace = namespace
+
+	testArn := "arn:aws:iam::123456789012:role/test_role"
+	svcAccount.Annotations = make(map[string]string)
+	svcAccount.Annotations[iam4kube.IamRoleArnAnnotation] = testArn
+	kroler.svcAccIdx.Add(&svcAccount)
+
+	expectedRole := &iam4kube.IamRole{}
+	expectedRole.Arn = arn.ARN{"aws", "iam", "", "123456789012", "role/test_role"}
+	expectedRole.SessionName = namespace + "/" + name
+
+	role, err := kroler.RoleForIp(context.Background(), iam4kube.IP(ipAddr))
+	assert.Nil(t, err)
```

Comment:
`require.NoError(t, err)`. `require.*` methods abort test execution on failure by unwinding the stack, executing all `defer`ed dunctions. I.e. you don't want to proceed if `err` is not `nil`.  And `NoError` will give a more meaningful message on failure.